### PR TITLE
Serviceaccounts: usage statistics

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
Currently we have no way of knowing if someone is running serviceaccounts.
This implementation gives a way to export metrics such as how much serviceaccounts endpoint is used.

**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana/issues/42328

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/serviceaccounts/database/metrics.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+//Counts the number of rows in a table.  Only use trusted strings.
+func (sa *ServiceAccountsStoreImpl) CountTable(table string) int {
+	var result int = -1
+	_ = sa.sqlStore.WithDbSession(context.TODO(), func(sess *sqlstore.DBSession) error {
+		_, err := sess.SQL("SELECT count(1) FROM " + table + " where is_service_account = true").Get(&result)
+		return err
+	})
+	return result
+}
```
Comment (donomii) [inline pkg/services/serviceaccounts/database/metrics.go:63]: I'm regretting writing this function.  Can we "sanitise" the table string before we use it?

File: pkg/services/serviceaccounts/manager/service.go [hunk 2]
```diff
@@ -18,34 +20,63 @@ var (
...
 		sa.log.Debug(ServiceAccountFeatureToggleNotFound)
+		return errors.New(ServiceAccountFeatureToggleNotFound)
+	}
+	return nil
+}
+
+func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
+	err := sa.IsDisabled()
+	if err != nil {
 		return nil
```
Comment (donomii) [inline pkg/services/serviceaccounts/manager/service.go:63]:
```suggestion
		return errors.New(ServiceAccountFeatureToggleNotFound)
	}
	return nil
}

func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
	err := sa.IsDisabled()
	if err != nil {
		return err
```

**Comment by donomii** [comment]
Are you going to add some counters?

File: pkg/services/serviceaccounts/manager/service.go [hunk 3]
```diff
@@ -18,34 +20,63 @@ var (
...
+	})
+}
+
+func (sa *ServiceAccountsService) getUsageMetrics() interface{} {
+	err := sa.IsDisabled()
+	if err != nil {
+		return 0
+	}
+	return 1
+}
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/manager/service.go:82]: this is exposed to us! this is what want to get for now

Comment (donomii) [reply pkg/services/serviceaccounts/manager/service.go:82]: We could also count the number of accounts added, deleted, etc

File: pkg/services/serviceaccounts/database/metrics.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/services/serviceaccounts/database/metrics.go]: I will remove all of the metrics so far. Now let's just add usage metrics

**Pull request closed without merging**